src/main/java/org/broadinstitute/hellbender/tools/spark/sv/FindBadGenomicKmers.java

```diff
@@ -0,0 +1,135 @@
+
+    @Override
+    public boolean requiresReference()
+    {
+        return true;
+    }
+
+    // Get the list of high copy number kmers in the reference, and write them to a file.
+    @Override
+    protected void runTool( final JavaSparkContext ctx )
+    {
+        SAMFileHeader hdr = getHeaderForReads();
+        SAMSequenceDictionary dict = null;
+        if ( hdr != null ) dict = hdr.getSequenceDictionary();
+        List<Kmer> killList = findBadGenomicKmers(ctx,getReference(),dict);
+        final File killFile = new File(outputFile);
+        try
+        {
+            final Writer writer = new BufferedWriter(new OutputStreamWriter(new FileOutputStream(killFile)));
+
+            for ( Kmer kmer : killList )
+            {
+                writer.write(kmer.toString(KLEN));
+                writer.write('\n');
+            }
+            writer.close();
+        }
+        catch ( IOException e )
+        { throw new GATKException("Unable to write ubiquitous kmer kill file.", e); }
+    }
+
+    // Find high copy number kmers in the reference sequence
+    public static List<Kmer> findBadGenomicKmers( final JavaSparkContext ctx,
+                                                  final ReferenceMultiSource ref,
+                                                  final SAMSequenceDictionary readsDict )
+    {
+        // Write all the reference sequences to a text file as a series of 10,000-base lines.
+        // For each contig, records are written with a K-1 base overlap
+        // (i.e., the last 62 bases in line n match the first 62 bases in line n+1) so that we don't miss any kmers
+        // due to line breaks -- we can just kmerize each record.
+        File refSequenceFile;
+        try
+        {
+            refSequenceFile = File.createTempFile("genomic_sequence","txt");
+            refSequenceFile.deleteOnExit();
+            final OutputStream os = new BufferedOutputStream(new FileOutputStream(refSequenceFile));
+
+            final SAMSequenceDictionary dict = ref.getReferenceSequenceDictionary(readsDict);
+            if ( dict == null )
+                throw new GATKException("No reference dictionary available.");
+
+            final int effectiveRecLen = REF_REC_LEN - KLEN + 1;
+            for ( final SAMSequenceRecord rec : dict.getSequences() )
+            {
+                final String seqName = rec.getSequenceName();
+                final int seqLen = rec.getSequenceLength();
+                final byte[] bases = ref.getReferenceBases(null,new SimpleInterval(seqName,1,seqLen)).getBases();
+                for ( int start = 0; start < seqLen; start += effectiveRecLen )
+                {
+                    os.write(bases,start,Math.min(REF_REC_LEN,seqLen-start));
+                    os.write('\n');
+                }
+            }
+
+            os.close();
+        }
+        catch ( IOException e )
+        { throw new GATKException("Unable to write reference text file for kmerizing.", e); }
```

Comment:
It looks like this whole first half of the function should be extracted out into a new function `writeReferenceToTmpFile` or something along those lines